ipaclient/install/ipa_client_samba.py

```diff
@@ -0,0 +1,726 @@
+        try:
+            old_principal = getattr(context, "principal", None)
+            name = gssapi.Name(principal, gssapi.NameType.kerberos_principal)
+            store = {"ccache": ccache_file, "client_keytab": keytab}
+            gssapi.Credentials(name=name, usage="initiate", store=store)
+            # Finalize API when TGT obtained using host keytab exists
+            if not api.isdone("finalize"):
+                api.finalize()
+
+            # Now we have a TGT, connect to IPA
+            try:
+                if api.Backend.rpcclient.isconnected():
+                    api.Backend.rpcclient.disconnect()
+                api.Backend.rpcclient.connect()
+
+                yield
+            except gssapi.exceptions.GSSError as e:
+                raise Exception(
+                    "Unable to bind to IPA server. Error initializing "
+                    "principal %s in %s: %s" % (principal, keytab, str(e))
+                )
+        finally:
+            if api.Backend.rpcclient.isconnected():
+                api.Backend.rpcclient.disconnect()
+            setattr(context, "principal", old_principal)
+
+
+def parse_options():
+    usage = "%prog [options]\n"
+    parser = OptionParser(usage=usage)
+    parser.add_option("--server", dest="server", help="FQDN of IPA server")
+    parser.add_option(
+        "--netbios-name",
+        dest="netbiosname",
+        help="NetBIOS name of the machine",
+        default=None,
+    )
+    parser.add_option(
+        "--no-homes",
+        dest="no_homes",
+        action="store_true",
+        default=False,
+        help="Do not add [homes] share to the generated Samba configuration",
+    )
+    parser.add_option(
+        "--no-nfs",
+        dest="no_nfs",
+        action="store_true",
+        default=False,
+        help="Do not allow NFS integration (SELinux booleans)",
+    )
+    parser.add_option(
+        "--force",
+        dest="force",
+        action="store_true",
+        default=False,
+        help="force installation by redoing already done steps",
+    )
+    parser.add_option(
+        "--debug",
+        dest="debug",
+        action="store_true",
+        default=False,
+        help="enable debugging",
+    )
+    parser.add_option(
+        "-U",
+        "--unattended",
+        dest="unattended",
+        action="store_true",
+        default=False,
+        help="unattended installation never prompts the user",
+    )
+    parser.add_option(
+        "--uninstall",
+        dest="uninstall",
+        action="store_true",
+        default=False,
+        help="Revert configuration and remove SMB service",
+    )
+
+    options, args = parser.parse_args()
+    return options, args
+
+
+domain_information_template = """
+ Domain name: {domain_name}
+NetBIOS name: {netbios_name}
+         SID: {domain_sid}
+    ID range: {range_id_min} - {range_id_max}
+"""
+
+
+def pretty_print_domain_information(info):
+    result = []
+    for d in info:
```

Comment:
maybe domain instead of d? Or is it obvious?

Reply:
Turned to domain